src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -522,16 +556,18 @@ WorkerCreateShard(Oid relationId, char *nodeName, uint32 nodePort,
 						 WORKER_APPLY_INTER_SHARD_DDL_COMMAND, shardId, escapedSchemaName,
 						 referencedShardId, escapedReferencedSchemaName, escapedCommand);
 
-		queryResultList = ExecuteRemoteQuery(nodeName, nodePort, newShardOwner,
-											 applyForeignConstraintCommand);
-		if (queryResultList == NIL)
+		if (createInSeparateTransaction)
 		{
-			shardCreated = false;
-			break;
+			ExecuteRemoteQuery(nodeName, nodePort, newShardOwner,
```

Comment:
We want to phase out `ExecuteRemoteQuery` since it relies on legacy APIs and opens a new connection for every command (!). If you send a command over a connection (e.g. using `ExecuteCriticalRemoteCommand`) without having sent `BEGIN` then it will commit immediately.